### PR TITLE
Switch travis-ci to docker infrastructure

### PR DESCRIPTION
**Pull request by dagar**
Currently ninja-build isn't in the travis apt whitelist.

If this works we get ccache in travis!

### CODE REVIEW

**Comment by dagar** [comment]
Waiting on https://github.com/travis-ci/travis-ci/issues/3530 for a slightly newer cmake and https://github.com/travis-ci/travis-ci/issues/3529 for ninja.

**Comment by dagar** [comment]
With the docker infrastructure we should be able to easily cache the compiled LLVM for PR https://github.com/dropbox/pyston/pull/400.

**Comment by kmod** [comment]
cool, thanks for following up on this :)

**Comment by dagar** [comment]
Consistent failure in the gcc build pyston_experimental_pypa_parser. Is this named incorrectly now?
                  threading_local.py    FAILED (Exited with code -11 (expected code 0))

I'll disable the gcc build again.

**Comment by dagar** [comment]
@kmod this is working. Most builds should now be ~7-10 minutes when travis isn't acting up. The next step will be caching a fully built LLVM (PR #400) which should cut the remaining build time in half.

If you ever need to clear the cache - http://docs.travis-ci.com/user/caching/#Clearing-Caches

File: CMakeLists.txt [hunk 1]
```diff
@@ -188,7 +188,7 @@ add_test(NAME gc_unittest COMMAND gc_unittest)
 add_test(NAME analysis_unittest COMMAND analysis_unittest)
 add_test(NAME pyston_defaults COMMAND ${PYTHON_EXE} ${CMAKE_SOURCE_DIR}/tools/tester.py -R ./pyston -j${TEST_THREADS} -a=-S -k ${CMAKE_SOURCE_DIR}/test/tests)
 add_test(NAME pyston_max_compilation_tier COMMAND ${PYTHON_EXE} ${CMAKE_SOURCE_DIR}/tools/tester.py -R ./pyston -j${TEST_THREADS} -a=-O -a=-S -k ${CMAKE_SOURCE_DIR}/test/tests)
-add_test(NAME pyston_experimental_pypa_parser COMMAND ${PYTHON_EXE} ${CMAKE_SOURCE_DIR}/tools/tester.py -a=-x -R ./pyston -j${TEST_THREADS} -a=-n -a=-S -k ${CMAKE_SOURCE_DIR}/test/tests)
+add_test(NAME pyston_old_parser COMMAND ${PYTHON_EXE} ${CMAKE_SOURCE_DIR}/tools/tester.py -a=-x -R ./pyston -j1 -a=-n -a=-S -k ${CMAKE_SOURCE_DIR}/test/tests)
```
Comment (kmod) [inline CMakeLists.txt]: was this change to `-j1` intentional?

Comment (dagar) [inline CMakeLists.txt]: Yes, running -x in parallel fails intermittently. I didn't look into it because I assumed that mode would be gone eventually.

**Comment by dagar** [comment]
kubuntu-backports ppa (for cmake) now available for travis-ci

**Comment by kmod** [comment]
This looks good to me but the merge will have to wait until we can sort out the build breakages (sorry about that).

I don't think we should accept intermittent failures -- I'm adding some extra output to the tester which should hopefully make these things easier to debug.

**Pull request merged**